addons/binding/org.openhab.binding.opensprinkler/src/main/java/org/openhab/binding/opensprinkler/handler/OpenSprinklerHandler.java

```diff
@@ -0,0 +1,248 @@
+                    "Could not control the OpenSprinkler device.");
+            logger.error("Error controlling and OpenSprinkler station.");
+            logger.debug("Exception received: " + exp.toString());
+        }
+    }
+
+    @Override
+    public void initialize() {
+        openSprinklerConfig = getConfig().as(OpenSprinklerConfig.class);
+
+        logger.debug("Initializing OpenSprinkler Thing with config (Hostname: " + openSprinklerConfig.hostname
+                + ", Port: " + openSprinklerConfig.port + ", Password: " + openSprinklerConfig.password + ", Refresh: "
+                + openSprinklerConfig.refresh + ").");
+
+        if (openSprinklerConfig == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.CONFIGURATION_ERROR,
+                    "Could not parse the config for the OpenSprinkler thing.");
+            logger.error("Could not parse config for the OpenSprinkler device.");
+
+            return;
+        }
+
+        try {
+            openSprinklerDevice = OpenSprinklerApiFactory.getHttpApi(openSprinklerConfig.hostname,
+                    openSprinklerConfig.port, openSprinklerConfig.password);
+        } catch (Exception exp) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR,
+                    "Could not create a connection to the OpenSprinkler thing.");
+            logger.error("Could not create API connection to the OpenSprinkler device.");
+            logger.debug("Exception received: " + exp.toString());
+
+            return;
+        }
+
+        logger.debug("Successfully created API connection to the OpenSprinkler device.");
+
+        try {
+            openSprinklerDevice.openConnection();
+        } catch (Exception exp) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR,
+                    "Could not open the connection to the OpenSprinkler thing.");
+            logger.error("Could not open API connection to the OpenSprinkler device.");
+            logger.debug("Exception received: " + exp.toString());
+        }
+
+        if (openSprinklerDevice.isConnected()) {
+            updateStatus(ThingStatus.ONLINE);
+            logger.debug("OpenSprinkler connected.");
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR,
+                    "Could not initialize the connection to the OpenSprinkler thing.");
+            logger.error("Could not initialize the OpenSprinkler device.");
+
+            return;
+        }
+
+        onUpdate();
+    }
+
+    @Override
+    public void dispose() {
+        if (openSprinklerDevice != null) {
+            try {
+                openSprinklerDevice.closeConnection();
+            } catch (Exception exp) {
+                logger.error("Could not close API connection to the OpenSprinkler device.");

```

Comment:
on a dispose, the user will not really care - so reduce to debug level (i.e. merge with next line)